.github/workflows/tests.yml

```diff
@@ -26,7 +26,7 @@ jobs:
         run: "./scripts/ci/install-deps.sh --ignore-scripts"
 
       - name: Run tests with coverage
-        run: "yarn coverage --ci --silent > /dev/null 2>&1"
+        run: "yarn test --coverage --ci --reporters=jest-silent-reporter '--reporters=<rootDir>/test/reporter.js'"
```

Comment:
Can we please use the github-actions reporter like we do for js-sdk? This shows the failures in the diff tab. It also silences the output in the logs. https://github.com/matrix-org/matrix-js-sdk/blob/develop/.github/workflows/tests.yml#L29

Reply:
Okay, I'll use it for the "final" run 

Fyi, this PR is in very in a WIP stage, more a playground for analysis and improvement :)